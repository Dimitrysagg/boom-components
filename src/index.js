import * as colors from "./constants/colors";
import * as helpers from "./helpers";

export { helpers };
export { colors };

export { default as Badge } from "./components/data-display/badge";
export { default as Breadcrumb } from "./components/data-display/breadcrumb";
export { default as Card } from "./components/data-display/card";
export { default as Collapse } from "./components/data-display/collapse";
export { default as Drawer } from "./components/data-display/drawer";
export { default as Modal } from "./components/data-display/modal";
export {
  default as ProfilePicture
} from "./components/data-display/profile-picture";
export { default as Steps } from "./components/data-display/steps";

export { default as AutoComplete } from "./components/data-entry/autocomplete";
export { default as Checkbox } from "./components/data-entry/checkbox";
export { default as Dropdown } from "./components/data-entry/dropdown";
export { default as Input } from "./components/data-entry/input";
export { default as Radio } from "./components/data-entry/radio";
export { default as Slider } from "./components/data-entry/slider";
export { default as Switch } from "./components/data-entry/switch";

export { default as Alert } from "./components/feedback/alert";
export { default as Dialog } from "./components/feedback/dialog";
export { default as Message } from "./components/feedback/message";
export { default as Notification } from "./components/feedback/notification";
export { default as Placeholder } from "./components/feedback/placeholder";
export { default as Spin } from "./components/feedback/spin";
export { default as Tooltip } from "./components/feedback/tooltip";

export { default as Button } from "./components/general/button";
