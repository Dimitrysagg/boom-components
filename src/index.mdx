---
name: Boom Components
route: /
---

# Boom Components

Biblioteca de componentes React para criação de UIs nos aplicativos da Alboom.