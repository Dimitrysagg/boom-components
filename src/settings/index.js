/* eslint-disable no-underscore-dangle */
const scope = typeof window === "undefined" ? global : window;
const settings = scope.__boomComponentsSettings || {};
scope.__boomComponentsSettings = scope.__boomComponentsSettings || settings;

export const update = (name, config = {}) => {
  settings[name] = {
    ...config,
    ...(settings[name] || {})
  };

  return settings[name];
};

export const get = name => settings[name];
