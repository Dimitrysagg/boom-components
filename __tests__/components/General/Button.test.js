import React from 'react'
import { mount } from 'enzyme'

import { Button } from '../../../src/components'

describe('Button', () => {
  test('Snapshot', () => {
    const wrapper = mount(<Button>Test</Button>)
    const tree = wrapper.find(Button)

    expect(tree).toMatchSnapshot()
  })
})
