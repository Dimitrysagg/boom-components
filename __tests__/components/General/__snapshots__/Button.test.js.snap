// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot 1`] = `
<Button>
  <styled.button>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "hyTKFx",
            "rules": Array [
              "
  display: flex;
  align-items: center;
  justify-content: center;
  height: 35px;
  padding: 0 20px;
  border: 0;
  border-radius: 2px;
  font-size: 16px;
  cursor: pointer;
  transition: opacity 150ms ease-in-out;

  &:hover {
    opacity: 0.8;
  }

  &:disabled {
    cursor: no-drop;

    &:hover {
      opacity: 1;
    }
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <button
        className="sc-bdVaJa hyTKFx"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "gChKYK",
                  "rules": Array [
                    "
  flex: 1;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="sc-htpNat gChKYK"
            >
              Test
            </span>
          </StyledComponent>
        </styled.span>
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;
